File: Programmers/Lv2/HIndex/solution.py
def is_valid_h_index(citations: list, h: int) -> bool:
    target_idx = len(citations) - h
    return citations[target_idx] >= h


def solution(citations):
    citations.sort()

    for h in range(len(citations), 0, -1):
        if is_valid_h_index(citations, h):
            return h
    return 0

File: Programmers/Lv2/HIndex/Solution.java
package Lv2.HIndex;

import java.util.*;

class Solution {

    private boolean isValidHIndex(int[] arr, int h) {
        int idx = arr.length - h;
        return arr[idx] >= h;
    }

    public int solution(int[] citations) {
        Arrays.sort(citations);

        for (int h = citations.length; h >= 1; h--) {
            if (isValidHIndex(citations, h)) return h;
        }
        return 0;
    }

}